### PR TITLE
Deploy to PyPI via Trusted Publishers

### PR DESCRIPTION
**Pull request by hugovk**
For https://github.com/python/blurb/issues/4.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  - [ ] Click "Publish release"
+
+- [ ] Check the tagged [GitHub Actions build](https://github.com/python/blurb/actions/workflows/release.yml)
+      has deployed to [PyPI](https://pypi.org/project/blurb/#history)
+
+- [ ] Check installation:
+
+  ```bash
+  python -m pip uninstall -y blurb && python -m pip install -U blurb && blurb help
```
Comment (ezio-melotti) [inline RELEASING.md:28]: Should this include commands to set up a venv and possibly remove it afterwards?

Comment (hugovk) [reply RELEASING.md:28]: I don't think so, it's a CLI that needs be called from the CPython repo.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  # Publish to PyPI on GitHub Releases.
+  release-pypi:
+    name: Publish to PyPI
+    # Only run for published releases.
+    if: |
+      github.repository_owner == 'python'
+      && github.event.action == 'published'
+    runs-on: ubuntu-latest
+    needs: build-package
```
Comment (webknjaz) [inline .github/workflows/release.yml:60]: I strongly recommend adding an extra layer of protection — GitHub Environments. Among other things, setting a deployment target URL makes it show up nicely in a few places in the GitHub web UI:

```suggestion
    needs: build-package

    environment:
      name: pypi
      url: >-
        https://pypi.org/project/blurb/${{
          github.event.release.tag_name
        }}
```

You can optionally require a human to hit an approval button though this deployment environment's protection rules in the repo settings, making it the last line of defense.

*(@woodruffw and I are trying to get more tutorials to showcase always using this from the very beginning, especially the official ones)*

P.S. Also consider publishing to GH Releases with Sigstore-produced signatures as shown in my guide: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#signing-the-distribution-packages.

**Comment by webknjaz** [comment]
> Deploy to GitHub Actions

@hugovk it seems you meant to write "Deploy to PyPI" but made a typo, eh?

**Pull request merged**